production/helm/loki/templates/backend/service-backend-headless.yaml

```diff
@@ -35,6 +35,7 @@ spec:
       port: {{ .Values.loki.server.grpc_listen_port }}
       targetPort: grpc
       protocol: TCP
+      appProtocol: tcp
```

Comment:
Is the `appProtocol` really supposed to be `tcp`, not `kubernetes.io/h2c`?

https://github.com/grpc/grpc-go
https://kubernetes.io/docs/concepts/services-networking/service/#application-protocol

Reply:
I went with `tcp` is correct based on istio setup notes:
- Other services, example [read](https://github.com/grafana/loki/blob/main/production/helm/loki/templates/read/service-read-headless.yaml#L38)
- Istio [protocol selection docs](https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/#explicit-protocol-selection), lists the available options
- [Grafana docs](https://grafana.com/docs/loki/latest/setup/install/istio/#required-changes) on istio setup

Interestingly I don't think the grafana docs note any of the services that are run by backend pods so maybe there is something different needed here?

Reply:
Ok, approved to be consistent across read/write/backend